src/Samples/Basic/Temporal.Samples.DesignProcess/Part1_1_TypicalWorkflow.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Hosting;
+
+using Temporal.Common.DataModel;
+using Temporal.Worker.Activities;
+using Temporal.Worker.Hosting;
+using Temporal.Worker.Workflows;
+
+namespace Temporal.Sdk.BasicSamples
+{
+    public class Part1_1_TypicalWorkflow
+    {
+        [Workflow(runMethod: nameof(SayHelloAsync))]
+        public class SayHelloWorkflow
+        {
+            private const string AddresseeNameDefault = "Boss";
+
+            private string _addresseeName = null;
+
+            public async Task SayHelloAsync(WorkflowContext workflowCtx)
+            {
+                var greeting = new SpeechRequest($"Hello, {_addresseeName ?? AddresseeNameDefault}.");                
+                await workflowCtx.Orchestrator.Activities.ExecuteAsync("SpeakAGreeting", greeting);
+            }
+
+            [WorkflowSignalHandler]
+            public Task SetAddresseeAsync(SpeechRequest input, WorkflowContext _)
+            {
+                _addresseeName = input?.Text;
+                return Task.CompletedTask;                
+            }
+        }
+
+        /// <summary>
+        /// Parameters to workflow APIs (main method, signal & query parameters) and to activities must implement <see cref="IDataValue" />.
+        /// In some specialized cases where it is not possible, the raw (non-deserialized) payload may be accessed
+        /// directly (e.g. <see cref="Part4_1_BasicWorkflowUsage" /> and <see cref="Part4_2_BasicWorkflowUsage_MultipleWorkers" />).
+        /// </summary>
+        public class SpeechRequest : IDataValue
+        {
+            public SpeechRequest(string text)
+            {
+                Text = text;
+            }
+
+            public string Text
+            {
+                get; set;
+            }
+        }
+
+
+        /// <summary>A sample activity implementation.</summary>
+        public static class Speak
+        {
+            public static Task GreetingAsync(SpeechRequest input, WorkflowActivityContext activityCtx)
+            {
+                Console.WriteLine($"[{activityCtx.ActivityTypeName}] {input.Text}");
+                return Task.CompletedTask;
+            }
+        }
+
+        public static void Main(string[] args)
+        {
+            // Use the ASP.Net Core host initialization pattern:
+            // 'UseTemporalWorkerHost' will configure all the temporal defaults
+            // and also apply all the file-based application config files.
```

Comment:
File based? I think file-based configuration is out of scope for the SDKs at the moment.

Reply:
Very common for .NET web-API apps. Reading config from files is handled by the built-in API. Consuming it does not depend on where it originated.

Reply:
I agree with @cretz this is premature, let's nail the basics first